Fixes #4927 - Extend foreman-debug to look for extensions provided by pl...

bkearney opened the pull request:
...ugins.

This assumes a directory called /usr/share/foreman/debug. Any *.debug
file in there is sourced. The expectation is that plugins would put
files into this locations and can add files or commands as necessary

bkearney committed:
Fixes #4927 - Extend foreman-debug to look for extensions provided by plugins.

This assumes a directory called /usr/share/foreman/script/foreman-debug.d.
Any *.sh file in there is sourced. The expectation is that plugins would put
files into this locations and can add files or commands as necessary

bkearney commented:
This is what the katello debug plugin would look like as an examplar

https://github.com/bkearney/katello/commit/4d77563e9a9dc042ec702ba54cda9333897d9b0c

lzap commented:
Ok this is nice, I was planning to do something similar. Thanks. Couple of comments.
- I'd put this in script/foreman-debug.d for consistency (the same for foreman-tail.d which you can also extend if you like - https://github.com/theforeman/foreman/pull/1311)
- I would not give any extension to the files, if you need one then use .sh or .bash which gives us correct syntax highlighting in editors (but I'd source them all)
- we need to package this for both redhats and debians but I have good news - having Rakefile.dist this is easy - just follow the same pattern I used in foreman-tail (the link above)

Please provide a koji build once this is ready so we are sure it does not break packaging (it should not since script/ dir is copied recursively). You can use my "sbu" (stands for scratchbuild) script for that: https://github.com/lzap/bin-public/blob/master/sbu

Just do "sbu" in the foreman directory (local changes allowed!) to get a scratch build in our koji.

Once we have that, I can trigger debian scratch build too. It should also work just fine.

lzap commented:
Please ignore my packaging comment since these changes needs to be done in _katello_ plugin and not in foreman :-) I was referring to this fyi (not relevant): https://github.com/lzap/foreman/blob/foreman-tail-4659/Rakefile.dist#L50-L60

Sorry for the confusion, I think I need :bear: real quick.

lzap commented:
Apparently :beer: or better :beers: 

/me is out!

domcleal commented:
Jenkins tests our RPM builds, so that hasn't broken.  (Links via the 'rpm' sub-job if you dig.)

bkearney commented:
i would need to extend the spec file to build the directory.. I will make the changes lzap suggest and then send a pr for the spec file.

lzap commented:
But before I leave - I think we need to create that directory in the Rakefile, just do mkdir_p and that should be it!

bkearney commented:
where in the rakefile?

lzap commented:
https://github.com/lzap/foreman/blob/foreman-tail-4659/Rakefile.dist#L50-L60

Just do mkdir_p there, similar to the foreman-tail and we are fine. No need to add this to git I guess.

bkearney commented:
Updated based on your comments, and added changes to the Rakefile.dist file.

I also updated the equivilant katello plugin

https://github.com/bkearney/katello/commit/ebd1a12d67ead4d835dab399ab334142d1275bd8

lzap commented:
Excellent. This works for me! :+1:

domcleal commented on script/foreman-debug:
```diff
@@ -259,6 +259,14 @@ add_files /etc/{sysconfig,default}/libvirt*
 
+# Look for any debug extensions provided by plugins
+if [ -d "/usr/share/foreman/script/foreman-debug.d" ]; then
+    for extension in "/usr/share/foreman/script/foreman-debug.d/*.sh"; do
```
Nitpick, this file uses 2-space indentation.

domcleal commented:
@lzap if you've tested, go ahead & merge when the above nitpick's fixed

lzap commented:
Fixed the indentation.

Merged as 0c93b4620e4defb20ccaeb1e8381e1743a3a1fd2 for Foreman 1.5.
Thank you @bkearney!

## 

Later,

 Lukas "lzap" Zapletal
 irc: lzap #theforeman
